chasm/lib/callback/statemachine.go

```diff
@@ -0,0 +1,106 @@
+package callback
+
+import (
+	"time"
+
+	failurepb "go.temporal.io/api/failure/v1"
+	"go.temporal.io/server/chasm"
+	callbackspb "go.temporal.io/server/chasm/lib/callback/gen/callbackpb/v1"
+	"go.temporal.io/server/common/backoff"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+// EventScheduled is triggered when the callback is meant to be scheduled for the first time - when its Trigger
+// condition is met.
+type EventScheduled struct{}
+
+var TransitionScheduled = chasm.NewTransition(
+	[]callbackspb.CallbackStatus{callbackspb.CALLBACK_STATUS_STANDBY},
+	callbackspb.CALLBACK_STATUS_SCHEDULED,
+	func(mctx chasm.MutableContext, cb *Callback, event EventScheduled) error {
+		mctx.AddTask(cb, chasm.TaskAttributes{}, &callbackspb.InvocationTask{})
+		return nil
+	},
```

Comment:
nit: It's standard to call context variables `ctx`.

```suggestion
	func(ctx chasm.MutableContext, cb *Callback, event EventScheduled) error {
		ctx.AddTask(cb, chasm.TaskAttributes{}, &callbackspb.InvocationTask{})
		return nil
	},
```